tools/Docgen/Builder/Builder.cs

```diff
@@ -0,0 +1,57 @@
+using System.IO;
+using Docgen.Generators;
+
+namespace Docgen.Builder
+{
+    public static class Builder
+    {
+        public static void Build(DoxygenDatabase database)
+        {
+            CleanOutputDir(Options.Instance.OutputDirectory);
+            BuildDocs(database, Options.Instance.OutputDirectory);
+            BuildTableOfContents(database, Options.Instance.OutputDirectory);
+        }
+
+        private static void BuildTableOfContents(DoxygenDatabase database, string outputDirectory)
+        {
+            // Build the table of contents at {{outputDirectory}}/.toc/toc.md
```

Comment:
that is the toc just for the API docs right? how would this work together with the rest of the documentation?

Reply:
Yes and you can use the include macro in the main ToC.